src/clr-angular/utils/focus-trap/focus-trap.directive.spec.ts

```diff
@@ -12,50 +12,101 @@ import { NoopAnimationsModule } from '@angular/platform-browser/animations';
     beforeEach(() => {
       TestBed.configureTestingModule({ imports: [ClrFocusTrapModule], declarations: [TestComponent] });
-
       fixture = TestBed.createComponent(TestComponent);
-      component = fixture.componentInstance;
       fixture.detectChanges();
-      compiled = fixture.nativeElement;
-      directive = fixture.debugElement.query(By.directive(FocusTrapDirective)).injector.get(FocusTrapDirective);
 
+      component = fixture.componentInstance;
+      compiled = fixture.nativeElement;
       lastInput = compiled.querySelector('#last');
+
+      directiveDebugElement = fixture.debugElement.query(By.directive(FocusTrapDirective));
+      directiveElement = directiveDebugElement.nativeElement;
+      directiveInstance = directiveDebugElement.injector.get(FocusTrapDirective);
     });
 
     afterEach(() => {
       fixture.destroy();
     });
 
     it('should create directive', () => {
-      expect(directive).toBeTruthy();
+      expect(directiveInstance).toBeTruthy();
     });
 
     it('should add tabindex attribute with value zero', () => {
-      directive.ngAfterViewInit();
-      const element: HTMLElement = directive.elementRef.nativeElement;
-      expect(element.getAttribute('tabindex')).toEqual('0');
+      expect(directiveElement.getAttribute('tabindex')).toEqual('0');
     });
 
-    it(`should focus on trappable element when tab key is pressed and last input is active`, () => {
-      const element = directive.elementRef.nativeElement;
-      lastInput.focus();
-      directive.onFocusIn(tabEvent);
-      expect(document.activeElement).toEqual(element);
+    it('should add its off-screen focus rebounder elements to parent element on instantiation', () => {
+      const offScreenEls = document.body.querySelectorAll('span.offscreen-focus-rebounder');
+      expect(offScreenEls.length).toBe(2);
+      expect(offScreenEls[0].classList.contains('before-focus-trap')).toBeTruthy();
+      expect(offScreenEls[1].classList.contains('after-focus-trap')).toBeTruthy();
+    });
+
+    it('should rebound focus back to the directive if one of rebounding elements gets focused', () => {
+      const offScreenEls = document.body.querySelectorAll('span.offscreen-focus-rebounder');
+
+      const beforeRebound = offScreenEls[0] as HTMLElement;
+      const afterRebound = offScreenEls[1] as HTMLElement;
+
+      beforeRebound.focus();
+      expect(document.activeElement).toBe(directiveElement);
+      afterRebound.focus();
+      expect(document.activeElement).toBe(directiveElement);
+    });
+
+    it('should remove its off-screen focus rebounder elements from parent element on removal', () => {
+      let offScreenEls = document.body.querySelectorAll('span.offscreen-focus-rebounder');
+      expect(offScreenEls.length).toBe(2);
+      component.mainFocusTrap = false;
+      fixture.detectChanges();
+      offScreenEls = offScreenEls = document.body.querySelectorAll('span.offscreen-focus-rebounder');
```

Comment:
Double assignment, but nitpick on this line and line 76-77.
```
expect(document.body.querySelectorAll('span.offscreen-focus-rebounder').length).toBe(0);
```

Reply:
Addressed this.